classes/CartRule.php

```diff
@@ -1402,6 +1402,48 @@ public function checkProductRestrictionsFromCart(CartCore $cart, $returnProducts
         return (!$displayError) ? true : false;
     }
 
+    protected function hasHigherPriorityThan(CartRule $otherCartRule): bool
+    {
+        // Compare by priority field (lower number = higher priority)
+        if ($this->priority !== $otherCartRule->priority) {
+            return $this->priority < $otherCartRule->priority;
+        }
+
+        // If priorities are equal, compare by discount type
+        $thisTypeWeight = $this->getTypeWeight();
+        $otherTypeWeight = $otherCartRule->getTypeWeight();
+
+        if ($thisTypeWeight !== $otherTypeWeight) {
+            return $thisTypeWeight < $otherTypeWeight;
+        }
+
+        // If types are also equal, compare by creation date (older = higher priority)
+        $thisDate = strtotime($this->date_add);
+        $otherDate = strtotime($otherCartRule->date_add);
+
+        return $thisDate < $otherDate;
+    }
+
+    protected function getTypeWeight(): int
+    {
+        $type = $this->getType();
+
+        if ($type === null) {
+            // If no type is defined, use a default weight (lowest priority)
+            return 999;
+        }
+
+        $typeWeights = [
+            DiscountType::PRODUCT_LEVEL => 1,
```

Comment:
Just a comment, we can maybe store these weight values as constants (reusable elsewhere).